components/script.tsx

```diff
@@ -78,7 +79,7 @@ const Script: React.FC<ScriptProps> = ({ className, messagesHeight = 'h-full', e
 
         let threadId = "";
         if (hasNoUserMessages() && enableThreads && !thread && setThreads && setSelectedThreadId) {
-            const newThread = await createThread(script, message, scriptId);
+            const newThread = await createThread(script, message, scriptId, workspace);
```

Comment:
Here when we create the thread, we check if workspace has been changed and if so, use the changed workspace to create thread